RecoLocalCalo/HcalRecAlgos/interface/AbsHBHEPhase1Algo.h

```diff
@@ -0,0 +1,56 @@
+#ifndef RecoLocalCalo_HcalRecAlgos_AbsHBHEPhase1Algo_h_
+#define RecoLocalCalo_HcalRecAlgos_AbsHBHEPhase1Algo_h_
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
+#include "DataFormats/HcalRecHit/interface/HBHEChannelInfo.h"
+#include "CalibFormats/HcalObjects/interface/HcalCalibrations.h"
+#include "CondFormats/HcalObjects/interface/HcalRecoParam.h"
+
+class AbsHFPhase1AlgoData;
+
+//
+// It is assumed that the HBHE Phase 1 algorithms will be developed
+// utilizing pairs of classes: a DB class with relevant calibration
+// constants and configuration parameters, derived from AbsHFPhase1AlgoData,
+// and the algo class, derived from AbsHBHEPhase1Algo, utilizing that DB class.
+//
+// We can expect that the same calibration constants might be utilized
+// by different reco algorithms.
+//
+// In principle, of course, the configuration objects do not have to be
+// actually stored in the database. It is expected that, at the early
+// stages of our understanding of energy reconstruction with QIE11
+// ASICs, these objects will be created from the module configuration
+// parameters.
+//
+class AbsHBHEPhase1Algo
+{
+public:
+    inline virtual ~AbsHBHEPhase1Algo() {}
+
+    inline virtual void beginRun(const edm::Run&, const edm::EventSetup&) {}
+    inline virtual void endRun() {}
+
+    // Does this class expect to receive its configuration from the database?
+    virtual bool isConfigurable() const = 0;
+
+    // If using DB, expect that the configuration will be updated
+    // once per run. We will not manage the pointer here. "true"
+    // should be returned on success (typically, automatic cast
+    // from the pointer checked by the appropriate dynamic cast).
+    inline virtual bool configure(const AbsHFPhase1AlgoData*) {return false;}
```

Comment:
Yes, it is possible. I would prefer to do that when developing
something else in CondFormats/HcalObjects. Approving any
change there takes a long time.

On 07/01/2016 05:42 AM, Slava Krutelyov wrote:
>
> In RecoLocalCalo/HcalRecAlgos/interface/AbsHBHEPhase1Algo.h <https://github.com/cms-sw/cmssw/pull/15092#discussion_r69281755>:
>
> > +class AbsHBHEPhase1Algo
> > +{
> > +public:
> > +    inline virtual ~AbsHBHEPhase1Algo() {}
> > +
> > +    inline virtual void beginRun(const edm::Run&, const edm::EventSetup&) {}
> > +    inline virtual void endRun() {}
> > +
> > +    // Does this class expect to receive its configuration from the database?
> > +    virtual bool isConfigurable() const = 0;
> > +
> > +    // If using DB, expect that the configuration will be updated
> > +    // once per run. We will not manage the pointer here. "true"
> > +    // should be returned on success (typically, automatic cast
> > +    // from the pointer checked by the appropriate dynamic cast).
> > +    inline virtual bool configure(const AbsHFPhase1AlgoData*) {return false;}
>
> is it still possible to rename AbsHFPhase1AlgoData to AbsHcalPhase1AlgoData ?
> Semantically, configuration of HBHE by HF config data sounds odd.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15092/files/6d2850b92942e3626e740059aff5348d603d1242#r69281755>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AFS4-WnHHFihNWMcPc3EDD1Q-ib9p0ftks5qRO8rgaJpZM4JCc97>.
>